src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -3122,7 +3122,7 @@ void G1PrintRegionLivenessInfoClosure::log_cset_candidate_group_add_total(G1CSet
                           group->group_id(),
                           group->length(),
                           group->gc_efficiency(),
-                          group->liveness_percent(),
+                          group->length() > 0 ? group->liveness_percent() : 0.0f,
```

Comment:
Why not move this logic into `liveness_percent()` (i.e returns 0.0f if length() == 0)?

Reply:
I do not feel strongly about this, and I can see both sides. Changed it to do the check in the logging directly.

For simplicity I want to avoid a "n/a" or something, and in fact we recently changed it to just return 0.